ExploreASL_ImportWorkflow.m

```diff
@@ -0,0 +1,123 @@
+function [x] = ExploreASL_ImportWorkflow(x)
+%ExploreASL_ImportWorkflow Multi-step import workflow for DCM2NII, NII2BIDS & BIDS2LEGACY.
+%
+% FORMAT: [x] = ExploreASL_ImportWorkflow(x)
+%
+% INPUT:
+%   x             - Struct containing pipeline environment parameters, useful when only initializing ExploreASL/debugging
+%
+% OUTPUT:
+%   x             - Struct containing pipeline environment parameters, useful when only initializing ExploreASL/debugging
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:    Multi-step import workflow for DCM2NII, NII2BIDS & BIDS2LEGACY.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:        n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Import Workflow
+    
+    % We expect x.DataParPath to be the sourceStructure.json file, so the root directory should be the study directory
+    [x.StudyRoot,~,~] = xASL_fileparts(x.DataParPath);
```

Comment:
I was almost waiting for feedback on this :)
I just thought it would be a good idea style-wise to create a separate variable for this, but maybe there is already a correct one in the x-struct. `ExploreASL_ImportBIDS` needs the folder containing the sourcedata, rawdata etc. which is the folder of the sourceStructure, which is why I defined this field. We can alternatively just remove it from the x-struct or put it in one of its fields, I guess, just let me know which one.

Reply:
It's good to think this a bit through. "DataPar" contained settings about data, this is not allowed in BIDS, and we don't need this anymore as all these settings are now in the json sidecars. So all we need still is a settings par file (which can still be called DataPar for backward compatibility). 

I'm also wondering if we shouldn't restructure the fieldnaming in the x struct while we're at it.
At least we should keep environmental (current choice, potentially platform-dependent) and data parameters (always the same, belong to data) separately. Also Pieter Vandemaele had good ideas about this, and drafted this in his docs on Google Drive

Reply:
Sure, just tell me what you want it to look like. For the parsing it's just quite useful to have all the arguments in the x struct, but I don't care that much about the field. In python people usually have `opts` (options) for the input arguments as a real only structure. The input parser does the same, you get p.results as read only. So I think we could use something like ...
```
x.opts.SkipPause
x.opts.iWorker
x.opts.nWorkers
```
... for the semi-important arguments and then we use ...
```
x.data.StudyRoot
x.data.DataParPath
```
... and maybe a separate processing ...
```
x.proc.ImportArray
x.proc.ProcessArray
```
I just want to make progress, so let me know what you'd like so I can go on implementing.

Reply:
Okay, I added everything to the dev meeting notes. We can discuss this there and update it after the #290 merge in a separate issue.